localstack/services/cloudformation/engine/template_preparer.py

```diff
@@ -29,21 +37,156 @@ def template_to_json(template: str) -> str:
     return json.dumps(template)
 
 
-def transform_template(template_body: str) -> str | None:
+def transform_template(template: Dict, parameters: List) -> Dict:
+    result = dict(template)
+
+    transformations = format_transforms(result.get("Transform", []))
+    for transformation in transformations:
+        if not isinstance(transformation["Name"], str):
+            # TODO this should be done during template validation
+            raise CommonServiceException(
+                code="ValidationError",
+                status_code=400,
+                message="Key Name of transform definition must be a string.",
+                sender_fault=True,
+            )
+        elif transformation["Name"] == SERVERLESS_TRANSFORM:
+            result = apply_serverless_transformation(result)
+        elif transformation["Name"] == EXTENSIONS_TRANSFORM:
+            continue
+        else:
+            result = execute_macro(
+                parsed_template=result,
+                macro=transformation,
+                stack_parameters=parameters,
+            )
+
+    return result
+
+
+def execute_macro(parsed_template: Dict, macro: Dict, stack_parameters: List) -> str:
+    macro_definition = get_cloudformation_store().macros.get(macro["Name"])
+    if not macro_definition:
+        raise FailedTransformation(macro["Name"], "2DO")
+
+    formatted_stack_parameters = {
+        param["ParameterKey"]: param["ParameterValue"] for param in stack_parameters
+    }
+
+    formatted_transform_parameters = macro.get("Parameters", {})
+    for k, v in formatted_transform_parameters.items():
+        if isinstance(v, Dict) and "Ref" in v:
+            formatted_transform_parameters[k] = formatted_stack_parameters[v["Ref"]]
+
+    transformation_id = f"{get_aws_account_id()}::{macro['Name']}"
+    event = {
+        "region": aws_stack.get_region(),
```

Comment:
```suggestion
        "region": aws_stack.get_region(),
```

We'll need to fix this later. Not important right now though since the deployer doesn't really support cross-region or cross-account anyway.